src/cc/bcc_syms.cc

```diff
@@ -416,50 +416,68 @@ static int _find_sym(const char *symname, uint64_t addr, uint64_t,
   return 0;
 }
 
+struct load_addr_t {
+  uint64_t target_addr;
+  uint64_t binary_addr;
+};
+int _find_load(uint64_t v_addr, uint64_t mem_sz, uint64_t file_offset,
+                       void *payload) {
+  struct load_addr_t *addr = static_cast<load_addr_t *>(payload);
+  if (addr->target_addr >= v_addr && addr->target_addr < (v_addr + mem_sz)) {
+    addr->binary_addr = addr->target_addr - v_addr + file_offset;
+    return -1;
+  }
+  return 0;
+}
+
 int bcc_resolve_symname(const char *module, const char *symname,
                         const uint64_t addr, int pid,
                         struct bcc_symbol_option *option,
                         struct bcc_symbol *sym) {
-  uint64_t load_addr;
   static struct bcc_symbol_option default_option = {
     .use_debug_file = 1,
     .check_debug_file_crc = 1,
     .use_symbol_type = BCC_SYM_ALL_TYPES,
   };
 
-  if (module == NULL)
-    return -1;
-
   memset(sym, 0, sizeof(bcc_symbol));
 
+  if (module == NULL)
+    return -1;
```

Comment:
Any particular reason why you move if condition here?

Reply:
Well, I think it would be good to do `memset` on result before returning from error

Reply:
The return value -1 should already indicate the error. The user should not count on any particular "sym" values. BTW, this is per user space, there is no security concerns.